src/main/webapp/components/TopNavigationMenu.js

```diff
@@ -114,31 +113,27 @@ const TopNavigationMenu = hh(class TopNavigationMenu extends Component {
 
   onSuccess = async (token) => {
     await User.signIn(token);
-    Storage.setUserIsLogged(true);
     this.init().then(resp => {
       if (Storage.getLocationFrom() != null) {
         this.props.history.push(Storage.getLocationFrom());
-        Storage.removeLocationFrom();
       } else {
         this.props.history.push("/index");
       }
-      window.location.reload();
+      Storage.removeLocationFrom();
     });
   };
 
   signOut = async () => {
-    this.props.showSpinner();
+    await User.signOut();
     Storage.clearStorage();
-    if (window.gapi.auth2 != undefined) {
-      let auth2 = window.gapi.auth2.getAuthInstance();
-      await auth2.signOut();
-      await User.signOut();
-    }
-    this.setState({
+    component.isBroad = null;
+    component.isAdmin = null;
+    component.isViewer = null;
```

Comment:
I'm seeing this in a lot of components. We should look into refactoring these into a user check or into a common HOC.